### PR TITLE
Chore/sql editor 2

### PR DESCRIPTION
**Pull request by saltcod**
— adds a new card template for creating a View
— persists the editor/results split screen size to localStorage
— Shows the dropdown caret for SQL Snippets on hover instead of having to click to make it active first

CleanShot 2023-06-19 at 14 31 13@2x
CleanShot 2023-06-19 at 14 32 24@2x

### CODE REVIEW

File: studio/components/layouts/SQLEditorLayout/QueryItem.tsx [hunk 1]
```diff
@@ -129,14 +136,16 @@ const QueryItemActions = observer(({ tabInfo }: { tabInfo: SqlSnippet }) => {
             </>
           }
         >
-          <Button
-            asChild
-            type="text"
-            icon={<IconChevronDown size="tiny" strokeWidth={2} className="text-scale-1100" />}
-            style={{ padding: '3px' }}
+          <span
```
Comment (MildTomato) [inline studio/components/layouts/SQLEditorLayout/QueryItem.tsx:139]:
```js
<Button asChild> // remember to have this as a Button
<span
  className={clsx(
   "[data-state='open']:bg-scale-400",
   "[data-state='open']:text-scale-900",
   "[data-state='open']:[_svg]:stroke-scale-900",
   //.. and whatever else you want to do
  )}
>
  <IconChevronDown size="tiny" strokeWidth={2} />
</span>
</Button>
```

Comment (MildTomato) [reply studio/components/layouts/SQLEditorLayout/QueryItem.tsx:139]: I think you can also do 

`"data-[state=open]:bg-scale-400"` as well

File: studio/components/layouts/SQLEditorLayout/QueryItem.tsx [hunk 2]
```diff
@@ -38,28 +38,34 @@ const QueryItem = ({ tabInfo }: QueryItemProps) => {
...
 export default QueryItem
 
-const QueryItemActions = observer(({ tabInfo }: { tabInfo: SqlSnippet }) => {
+interface QueryItemActionsProps {
+  tabInfo: SqlSnippet
+  activeId: string | undefined
+}
+
+const QueryItemActions = observer(({ tabInfo, activeId }: QueryItemActionsProps) => {
+  console.log('activeId', activeId)
```
Comment (joshenlim) [inline studio/components/layouts/SQLEditorLayout/QueryItem.tsx:68]:
```suggestion
```

File: studio/components/layouts/SQLEditorLayout/QueryItem.tsx [hunk 3]
```diff
@@ -38,28 +38,34 @@ const QueryItem = ({ tabInfo }: QueryItemProps) => {
...
-        isActive && 'text-scale-1200 bg-scale-400 dark:bg-scale-300 -active'
+        isActive && 'text-scale-1200 bg-scale-400 dark:bg-scale-600 -active'
       )}
       ref={isActive ? (activeItemRef as React.RefObject<HTMLDivElement>) : null}
     >
       <Link href={`/project/${ref}/sql/${id}`}>
         <a className="py-1 px-3 w-full">
           <p
-            title={name}
+            title={description}
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply studio/components/layouts/SQLEditorLayout/QueryItem.tsx:48]: this is for accessibility

File: studio/components/layouts/SQLEditorLayout/QueryItem.tsx [hunk 4]
```diff
@@ -38,28 +38,34 @@ const QueryItem = ({ tabInfo }: QueryItemProps) => {
       key={id}
```
Comment (MildTomato) [inline studio/components/layouts/SQLEditorLayout/QueryItem.tsx:38]: future reference.. could be done like this now...


```
<Link href={`/project/${ref}/sql/${id}`} key={id}>
<Button 
      asChild
      className={clsx(
        'flex items-center justify-between rounded-md group',
        isActive && 'text-scale-1200 bg-scale-400 dark:bg-scale-600 -active'
      )}
      ref={isActive ? (activeItemRef as React.RefObject<HTMLDivElement>) : null} // not sure you'll need this anymore?
    >
        <a className="py-1 px-3 w-full">

        // the rest...
        ```

**Review by MildTomato — APPROVED** [review]

**Pull request merged**